base_local_planner/include/base_local_planner/goal_functions.h

```diff
@@ -102,6 +102,7 @@ namespace base_local_planner {
       const geometry_msgs::PoseStamped& global_robot_pose,
       const costmap_2d::Costmap2D& costmap,
       const std::string& global_frame,
+      const ros::Time& time,
```

Comment:
I've spent quite a bit of time looking at this now - I really don't like littering extra params all over, and having all this book-keeping going on as to what time we are using for things (it seems like this will be very easy to use incorrectly).